core/src/thread_mem_usage.rs

```diff
@@ -37,3 +42,49 @@ impl Allocatedp {
         0
     }
 }
+
+#[derive(PartialEq, Debug, Serialize, Deserialize)]
+struct MapFile(HashMap<String, u64>);
+
+impl MapFile {
+    pub fn write_to_file(&self, path: &Path) {
+        let mut file = File::create(path).expect("Failed to create / write a file.");
+        let str = serde_json::to_string_pretty(self).expect("Error serializing the file.");
+        if let Err(err) = file.write_all(str.as_bytes()) {
+            panic!("Failed to write a file {}", err);
+        }
+    }
+
+    pub fn from_file(path: &Path) -> Self {
+        let mut file = File::open(path).expect("Could not open a file.");
+        let mut content = String::new();
+        file.read_to_string(&mut content)
+            .expect("Could not read from key file.");
+        serde_json::from_str(&content).expect("Failed to deserialize KeyFile")
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_thread_memory_map() {
+        let adp = Allocatedp::default();
+        let mem = adp.get();
+
+        let current_thread = thread::current();
+        let name = current_thread.name().unwrap();
+
+        let mut map: HashMap<String, u64> = HashMap::new();
+        map.insert(name.to_string(), mem);
+
+        let file = MapFile(map);
+        let path = Path::new("file.txt");
```

Comment:
Can your test put the file into `farf` and also cleanup the file in the case there is an error.

Reply:
it removes a file in a default case. Does it sound good?